src/Functions/FunctionMathUnary.h

```diff
@@ -154,6 +154,8 @@ class FunctionMathUnary : public IFunction
             using ColVecType = ColumnVectorOrDecimal<Type>;
 
             const auto col_vec = checkAndGetColumn<ColVecType>(col.column.get());
+            if (col_vec == nullptr)
```

Comment:
Will prevent the crash but the true problem is further up the stack, I guess.

\+ needs a test

Reply:
Currently building binary to canonize the test. But I agree that we need to find root cause.